src/coreclr/utilcode/pedecoder.cpp

```diff
@@ -1039,7 +1039,8 @@ CHECK PEDecoder::CheckCorHeader() const
     // If the file is COM+ 1.0, which by definition has nothing the runtime can
     // use, or if the file requires a newer version of this engine than us,
     // it cannot be run by this engine.
-    CHECK(VAL16(pCor->MajorRuntimeVersion) > 1 && VAL16(pCor->MajorRuntimeVersion) <= COR_VERSION_MAJOR);
+    // TODO: WIP composite r2r violate this, perhps should be fixed.
+    // CHECK(VAL16(pCor->MajorRuntimeVersion) > 1 && VAL16(pCor->MajorRuntimeVersion) <= COR_VERSION_MAJOR);
```

Comment:
I wonder if these violations can be fixed in the whatever produces r2r files. Currently this is set to `0` and hits this assert. There is a similar case below.